alchemist-engine/src/main/java/it/unibo/alchemist/core/ArrayIndexedPriorityFixedBatchQueue.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2010-2023, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.core;
+
+import it.unibo.alchemist.model.Actionable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Batched extension for ArrayIndexedPriorityQueue.
+ * This implementation presents fixed size batches.
+ *
+ * @param <T> concentration type
+ */
+public final class ArrayIndexedPriorityFixedBatchQueue<T> extends ArrayIndexedPriorityQueue<T> implements BatchedScheduler<T> {
```

Comment:
I would rewrite this in kotlin as

```suggestion
class ArrayIndexedPriorityFixedBatchQueue<T> : Scheduler<T> by ArrayIndexedPriorityQueue<T>(), BatchedScheduler<T> {
```

Reply:
Done